crates/sqlexec/src/dispatch/external.rs

```diff
@@ -148,160 +145,282 @@ impl<'a> ExternalDispatcher<'a> {
-                    table: name.to_owned(),
-                };
-                let prov = PostgresTableProvider::try_new(prov_conf).await?;
-                Ok(Arc::new(prov))
-            }
-            DatabaseOptions::BigQuery(DatabaseOptionsBigQuery {
-                service_account_key,
-                project_id,
-            }) => {
-                let table_access = BigQueryTableAccess {
-                    dataset_id: schema.to_string(),
-                    table_id: name.to_string(),
-                };
-
-                let accessor =
-                    BigQueryAccessor::connect(service_account_key.clone(), project_id.clone())
-                        .await?;
-                let provider = accessor.into_table_provider(table_access, true).await?;
-                Ok(Arc::new(provider))
-            }
-            DatabaseOptions::Mysql(DatabaseOptionsMysql { connection_string }) => {
-                let table_access = MysqlTableAccess {
-                    schema: schema.to_string(),
-                    name: name.to_string(),
-                };
-
-                let accessor = MysqlAccessor::connect(connection_string, tunnel).await?;
-                let provider = accessor.into_table_provider(table_access, true).await?;
-                Ok(Arc::new(provider))
-            }
-            DatabaseOptions::MongoDb(DatabaseOptionsMongoDb {
-                connection_string, ..
-            }) => {
-                let table_info = MongoDbTableAccessInfo {
-                    database: schema.to_string(),
-                    collection: name.to_string(),
-                    fields: None,
-                };
-                let accessor = MongoDbAccessor::connect(connection_string).await?;
-                let table_accessor = accessor.into_table_accessor(table_info);
-                let provider = table_accessor.into_table_provider().await?;
-                Ok(Arc::new(provider))
-            }
-            DatabaseOptions::Snowflake(DatabaseOptionsSnowflake {
-                account_name,
-                login_name,
-                password,
-                database_name,
-                warehouse,
-                role_name,
-            }) => {
-                let schema_name = schema.to_string();
-                let table_name = name.to_string();
-                let role_name = if role_name.is_empty() {
-                    None
-                } else {
-                    Some(role_name.clone())
-                };
+        // TODO, use the DATASOURCE_REGISTRY to dispatch the database.
 
-                let conn_params = SnowflakeDbConnection {
-                    account_name: account_name.clone(),
-                    login_name: login_name.clone(),
-                    password: password.clone(),
-                    database_name: database_name.clone(),
-                    warehouse: warehouse.clone(),
+        if let Some(ds) = DATASOURCE_REGISTRY.get(&db.options.as_str()) {
```

Comment:
I'm not sure I understand the goal/path here: having this match statement _here_ still has the same dependency properties as it did before. Is there more work planned here?

Reply:
The `else` statement will eventually go away once everything is migrated to the new table options. The goal here is _decoupling_. Now everything is encapsulated within the `DATASOURCE_REGISTRY` and the external dispatcher is no longer tightly coupled to the individual variants of the datasources. Right now the `DATASOURCE_REGISTRY` is a static global, but the idea long term is to do the same as we did with the `FunctionRegistry` in https://github.com/GlareDB/glaredb/pull/2699 which will allow us to dynamically register new datasources.